lang/stata/stata.py

```diff
@@ -0,0 +1,170 @@
+from talon import Context, actions, settings
+
+ctx = Context()
+
+ctx.matches = r"""
+code.language: stata
+"""
+
+# functions.py
+ctx.lists["user.code_parameter_name"] = {
+    # regressions
+    "v c e cluster": "vce(cluster)",
+    "v c e robust": "vce(robust)",
+}
+
+# functions_common.py
+ctx.lists["user.code_common_function"] = {
+    # base stata
+    "global": "global",
+    "local": "local",
+    "reg": "reg",
+    "regress": "reg",
+    # packages
+    "estadd": "estadd",
+    "estout": "estout",
+    "estpost": "estpost",
+    "eststo": "eststo",
+    "esttab": "esttab",
+}
+
+# libraries_gui.py
+ctx.lists["user.code_libraries"] = {
+    "estout": "estout",
+}
+
+
+@ctx.action_class("user")
+class UserActions:
+    # comment_line.py
+    def code_comment_line_prefix():
+        actions.auto_insert("* ")
+        # actions.auto_insert("// ")
+
+    # functions.py
+    def code_private_function(text: str):
+        result = "program {} \n\nend".format(
+            actions.user.formatted_text(
+                text, settings.get("user.code_private_function_formatter")
+            )
+        )
+        actions.user.paste(result)
+        actions.edit.up()
+        actions.key("tab")
+
+    def code_default_function(text: str):
+        actions.user.code_private_function(text)
+
+    def code_insert_named_argument(parameter_name: str):
+        actions.insert(f"{parameter_name} ")
+
+    # functions_common.py
+    def code_insert_function(text: str, selection: str):
+        text += f" {selection or ''}"
+        actions.user.paste(text)
+
+    # imperative.py
+    def code_block():
+        actions.auto_insert("\n")
+
+    def code_state_if():
+        actions.insert("if  {")
```

Comment:
Is this intentionally two spaces? It seems like below you use two after `if`'s, `else`'s, `foreach`, but one after `do {` and `for {`.

Reply:
I reworked the statements. They are now supposed to have two spaces so that after inserting a condition one does not need to add an extra space. With this reasoning the else statement does not need to spaces.